### ISSUE TITLE
cmake's C compiler test fails on arm

### ISSUE THREAD

**Issue by Grimler91** [termux/termux-packages#2983]
cmake's test for a working C compiler fails on arm android 7 (haven't tested other arches yet),
I think someone mentioned this on gitter the other day (ping @its-pointless).

Minimal example:
CMakeLists.txt
```
cmake_minimum_required(VERSION 3.5)
project (HELLO)
add_executable (hello hello.c)
```

hello.c
```
#include <stdio.h>
int main()
{
   printf("Hello, World!");
   return 0;
}
```

Running `cmake .` from the directory with the files outputs:
```
-- The C compiler identification is Clang 7.0.0
-- The CXX compiler identification is Clang 7.0.0
-- Check for working C compiler: /data/data/com.termux/files/usr/bin/cc
```
and then hangs forever. 

I've pinned down the problem to this try_compile function in `$PREFIX/share/cmake-3.12/Modules/CMakeTestCCompiler.cmake`:
```
  try_compile(CMAKE_C_COMPILER_WORKS ${CMAKE_BINARY_DIR}
    ${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/CMakeTmp/testCCompiler.c
    OUTPUT_VARIABLE __CMAKE_C_COMPILER_OUTPUT)
```

The `${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/CMakeTmp/testCCompiler.c` file is never created so I guess something goes wrong there

**Comment by its-pointless** [comment]
Its an issue with make as using ninja works.

**Comment by Grimler91** [comment]
You're right, I can confirm `cmake . -G Ninja -DCMAKE_MAKE_PROGRAM=$PREFIX/bin/ninja` works

**Comment by Grimler91** [comment]
I tried make without the latest commit: https://github.com/termux/termux-packages/commit/0acdf1ed946c5c279c1eb18637d1bb3358a54084 but that didn't change anything

**Comment by its-pointless** [comment]
compiling make with  ac_cv_func_pselect=no makes it work.

**Comment by its-pointless** [comment]
I got a proper fix in pull request.

**Comment by fornwall** [comment]
Should be fixed in version `4.2.1-2` of the make package, which is now available for installation. Nice work!
